JVM crashs in CI Builds

**Description**

It seems that we have again some SIGSEG issues, which causes to crash the JVM in our CI.

It downloaded the test errors zip from the following build https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/develop/821/ 
```
$ unzip test-errors-it.zip 
Archive:  test-errors-it.zip
  inflating: qa/integration-tests/hs_err_pid6859.log  
  inflating: qa/integration-tests/hs_err_pid6860.log  
$ cd qa/integration-tests/
$ ll
total 8
-rw-r--r-- 1 zell users 3144 Sep 16 15:13 hs_err_pid6859.log
-rw-r--r-- 1 zell users  396 Sep 16 15:10 hs_err_pid6860.log
$ cat hs_err_pid6860.log 
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f1975573435, pid=6860, tid=57545
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.10+9) (build 11.0.10+9)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.10+9, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [librocksdbjni4887027544215651946.so+0x395435][zell integration-tests/ cluster: tf-jx-precise-humpback ns:jx]$ 
```

```
$ cat hs_err_pid6859.log 
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f8e41d86821, pid=6859, tid=61320
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.10+9) (build 11.0.10+9)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.10+9, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libc.so.6+0x82821]
[timeout occurred during error reporting in step "printing problematic frame"] after 30 s.
# Core dump will be written. Default location: /core.%e.6859.%t
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  S U M M A R Y ------------

Command Line: -XX:+UseContainerSupport -XX:MaxRAMPercentage=8 -XX:MaxDirectMemorySize=4g -Djava.net.preferIPv4Stack=true --module-path=/home/jenkins/agent/workspace/camunda-cloud_zeebe_develop/qa/integration-tests/target/zeebe-qa-integration-tests-1.2.0-SNAPSHOT.jar --patch-module=zeebe.qa.integration.tests=/home/jenkins/agent/workspace/camunda-cloud_zeebe_develop/qa/integration-tests/target/test-classes --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.client=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.client.command=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.client.multipartition=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.clustering=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.clustering.topology=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.gateway=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.health=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.network=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.smoke=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.startup=ALL-UNNAMED --add-exports=zeebe.qa.integration.tests/io.camunda.zeebe.it.system=ALL-UNNAMED --add-modules=zeebe.qa.integration.tests --add-reads=zeebe.qa.integration.tests=ALL-UNNAMED org.apache.maven.surefire.booter.ForkedBooter /home/jenkins/agent/workspace/camunda-cloud_zeebe_develop/qa/integration-tests/target/surefire 2021-09-16T15-01-45_695-jvmRun1 surefire7038759527328167887tmp surefire_115532971887761201284tmp

Host: 
[timeout occurred during error reporting in step "printing summary machine and OS info"] after 30 s.
Time: Thu Sep 16 15:12:57 2021 UTC elapsed time: 401.604736 seconds (0d 0h 6m 41s)

---------------  T H R E A D  ---------------

Current thread (0x00007f8c4420c800):  JavaThread "Broker-2-zb-fs-workers-0" daemon [_thread_in_native, id=61320, stack(0x00007f8c07cbb000,0x00007f8c07dbc000)]

Stack: [0x00007f8c07cbb000,0x00007f8c07dbc000],  sp=0x00007f8c07db88b0,  free space=1014k
Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libc.so.6+0x82821]
------ Timeout during error reporting after 120 s. ------
```

[test-errors-it.zip](https://github.com/camunda-cloud/zeebe/files/7183048/test-errors-it.zip)

There is another reason why build fails, because of oom. https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/staging/1072/


```

[2021-09-17T07:15:52.183Z] [INFO] --- maven-failsafe-plugin:3.0.0-M5:verify (default) @ zeebe-qa-update-tests ---

[2021-09-17T07:16:04.608Z] [ERROR] OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00007f36ccaf5000, 16384, 0) failed; error='Not enough space' (errno=12)
[2021-09-17T07:16:04.608Z] [ERROR] OpenJDK 64-Bit Server VM warning: INFO: os::commit_memoOpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00007ff710503000, 65536, 1) failed; error='Not enough space' (errno=12)
[2021-09-17T07:16:04.608Z] ry(0x00007f87006f0000, 16384, 0) failed; error='Not enough space' (errno=12)
[2021-09-17T07:16:04.608Z] #
[2021-09-17T07:16:04.608Z] # There is insufficient memory for the Java Runtime Environment to continue.
[2021-09-17T07:16:04.608Z] # Native memory allocation (mmap) failed to map 65536 bytes for committing reserved memory.
[2021-09-17T07:16:04.608Z] # An error report file with more information is saved as:
[2021-09-17T07:16:04.608Z] # /home/jenkins/agent/workspace/camunda-cloud_zeebe_staging/hs_err_pid1384.log
[2021-09-17T07:16:04.608Z] #
[2021-09-17T07:16:04.608Z] # Compiler replay data is saved as:
[2021-09-17T07:16:04.608Z] # /home/jenkins/agent/workspace/camunda-cloud_zeebe_staging/replay_pid1384.log
[2021-09-17T07:16:04.608Z] + status=1
[2021-09-17T07:16:04.608Z] + analyseFlakyTests /tmp/tmp.aDJbOpA8qk ./FlakyTests.txt
[2021-09-17T07:16:04.608Z] + local -r testInputFile=/tmp/tmp.aDJbOpA8qk
[2021-09-17T07:16:04.608Z] + local -r testOutputFile=./FlakyTests.txt
[2021-09-17T07:16:04.608Z] ++ mktemp
[2021-09-17T07:16:04.608Z] + local -r irOutputFile=/tmp/tmp.bOIlTaeXyT
[2021-09-17T07:16:04.608Z] + grep -q '\[WARNING\] Flakes:' /tmp/tmp.aDJbOpA8qk
[2021-09-17T07:16:04.608Z] + return 0
[2021-09-17T07:16:04.608Z] + [[ 1 != 0 ]]
```

[test-errors-it (1).zip](https://github.com/camunda-cloud/zeebe/files/7183727/test-errors-it.1.zip)

The hs_error logs show more details.

--- a/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java
+++ b/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java
@@ -257,6 +257,8 @@ public final class ClusteringRule extends ExternalResource {
                 LOG.error("Failed to close broker: ", e);
               }
             });
+    systemContexts.values().forEach(ctx -> ctx.getScheduler().stop());
+    systemContexts.clear();
     brokers.clear();
     brokerCfgs.clear();
     logstreams.clear();

